Description: Fix tag_weight when using logarithmic distribution
 When logarithmic distribution is used _calculate_tag_weight may
 return weight greater than maximum threshold. Because of this the attribute
 font_size of the most frequently used tag may be left unset.
Upstream: http://code.google.com/p/django-tagging/issues/detail?id=91
Debian: #525770

--- django-tagging.old/tagging/utils.py	(revision 132)
+++ django-tagging/tagging/utils.py	(working copy)
@@ -230,7 +230,7 @@
     if distribution == LINEAR or max_weight == 1:
         return weight
     elif distribution == LOGARITHMIC:
-        return math.log(weight) * max_weight / math.log(max_weight)
+        return min((max_weight, math.log(weight) * max_weight / math.log(max_weight)))
     raise ValueError(_('Invalid distribution algorithm specified: %s.') % distribution)
 
 def calculate_cloud(tags, steps=4, distribution=LOGARITHMIC):

